Assets/Scripts/EnemyAI.cs

```diff
@@ -46,31 +54,35 @@ private void Start()
         lastFireTime = -fireRate;
         fireSfx = gunTip.GetComponent<AudioSource>();
         fireSfx.maxDistance = attackDistance * 2f;
-        alienAggroSfx = GetComponent<AudioSource>();
-        alienAggroSfx.maxDistance = chaseDistance * 2f;
+        var audioSources= GetComponents<AudioSource>();
+        aggroSfx = audioSources[0];
+        aggroSfx.maxDistance = chaseDistance * 2f;
+        screamSfx = audioSources[1];
         mainCamera = LevelManager.MainCamera.transform;
         thrusterParticleManager = GetComponentInChildren<ThrusterParticleManager>();
         projectileParent = GameObject.FindWithTag("ProjectileCollection").transform;
         entityManager = FindObjectOfType<ProceduralGeneration>();
+        grapple = player.GetComponentInChildren<GrappleGunBehavior>();
+        impulseVector = null;
     }
 
     private void Update()
     {
-        if (!ProceduralGeneration.FinishedGenerating) return;
+        if (!ProceduralGeneration.FinishedGenerating || currentState == State.Dead) return;
```

Comment:
Use LevelIsActive

Reply:
Nah because I want them to keep patrolling after you win or lose